relayer/chains/cosmos/cosmos_chain_processor.go

```diff
@@ -0,0 +1,56 @@
+package cosmos
+
+import (
+	"context"
+	"errors"
+
+	"github.com/cosmos/relayer/v2/relayer/processor"
+	"github.com/cosmos/relayer/v2/relayer/provider"
+	"github.com/cosmos/relayer/v2/relayer/provider/cosmos"
+	"go.uber.org/zap"
+)
+
+type CosmosChainProcessor struct {
+	ctx context.Context
+	log *zap.Logger
+
+	pathProcessors processor.PathProcessors
+	ChainProvider  *cosmos.CosmosProvider
+}
+
+func NewCosmosChainProcessor(ctx context.Context, log *zap.Logger, rpcAddress string, provider provider.ChainProvider, pathProcessors processor.PathProcessors) (*CosmosChainProcessor, error) {
+	cosmosProvider, ok := provider.(*cosmos.CosmosProvider)
+	if !ok {
+		return nil, errors.New("only cosmos providers are supported for cosmos chain processors")
+	}
```

Comment:
Why can't we accept the concrete type `*cosmos.CosmosProvider` as an argument vs `provider.ChainProvider`?

Reply:
> Why can't we accept the concrete type *cosmos.CosmosProvider as an argument vs provider.ChainProvider?
I was thinking it might make it easier for the caller so they don't have to cast, but I'm fine with making the API explicit instead of enforcing it here.